### PR TITLE
Adds a flag for scripts in script_tests.json

### PR DESCRIPTION
**Pull request by EthanHeilman**
changes script flags to use # instead of html like syntax

### CODE REVIEW

File: src/test/script_tests.cpp [hunk 1]
```diff
@@ -943,12 +943,15 @@ BOOST_AUTO_TEST_CASE(script_json_test)
...
                 auto element = test[pos][i].get_str();
-                // Taproot script - third from the last element in witness stack
-                if (test[pos].size() >= 3 && i == test[pos].size()-3 && !IsHex(element)) {
-                    witness.stack.push_back(ToByteVector(ParseScript(element)));
-                // Taproot script control block - second from the last element in witness stack
-                // If <AUTOGEN:CONTROLBLOCK> we auto-generate the control block
-                } else if (test[pos].size() >= 3 && i == test[pos].size()-2 && strcmp(element.c_str(), "<AUTOGEN:CONTROLBLOCK>") == 0) {
+                // We use #SCRIPT# to flag a non-hex script
+                std::string scriptFlag = std::string("#SCRIPT#");
+                if (element.find(scriptFlag) == 0) {
```
Comment (arminsabouri) [inline src/test/script_tests.cpp:948]: is interesting the `find` function returns 0 if there is a find. Or maybe it returns the index of the match

File: src/test/script_tests.cpp [hunk 2]
```diff
@@ -1925,8 +1928,10 @@ BOOST_AUTO_TEST_CASE(cat_dup_test)
             witVerifyScript.push_back(OP_CAT);
             int expectedErr = SCRIPT_ERR_OK;
             unsigned int catedStackElementSize = witData.at(0).size()*pow(2, dups);
-            if (catedStackElementSize > MAX_SCRIPT_ELEMENT_SIZE || elementSize > MAX_SCRIPT_ELEMENT_SIZE)
+            if (catedStackElementSize > MAX_SCRIPT_ELEMENT_SIZE || elementSize > MAX_SCRIPT_ELEMENT_SIZE){
                 expectedErr = SCRIPT_ERR_PUSH_SIZE;
+                break;
```
Comment (arminsabouri) [inline src/test/script_tests.cpp:1933]: can you remind me why we break here?

**Review by arminsabouri — APPROVED** [review]

File: src/test/script_tests.cpp (hunk 1 above)
Comment (EthanHeilman) [reply src/test/script_tests.cpp:948]: You are correct!

> The position of the first character of the first match.
 If no matches were found, the function returns [string::npos](https://cplusplus.com/string::npos).

https://cplusplus.com/reference/string/string/find/

File: src/test/script_tests.cpp (hunk 2 above)
Comment (EthanHeilman) [reply src/test/script_tests.cpp:1933]: The intention is break after we hit the first SCRIPT_ERR_PUSH_SIZE; error so that we don't waste time testing after we already past the edge case.  This was because my test was taking 40 seconds when rest the entire test suite was only taking 6 seconds.

However my code has a bug. It should be, the break hits too early

```cpp
for (unsigned int dups = 1; dups <= maxDupsToCheck; dups++) {
            witVerifyScript.push_back(OP_DUP);
            witVerifyScript.push_back(OP_CAT);
            int expectedErr = SCRIPT_ERR_OK;
            unsigned int catedStackElementSize = witData.at(0).size()*pow(2, dups);
            if (catedStackElementSize > MAX_SCRIPT_ELEMENT_SIZE || elementSize > MAX_SCRIPT_ELEMENT_SIZE){
                expectedErr = SCRIPT_ERR_PUSH_SIZE;
            }
            DoTapscriptTest(witVerifyScript, witData, "CAT DUP test", expectedErr);
            If ( expectedErr == SCRIPT_ERR_PUSH_SIZE)
                 break;
        }
```
